README.md

```diff
@@ -1,11 +1,24 @@
 ### Version numbers
 
-nanopb doesn't follow https://semver.org/, so we'll use the following scheme:
+Starting with nanopb 0.3.9.5, the CocoaPods version is disconnected from the
+nanopb version and is described by the following table:
 
-nanopb version `a.b.c` becomes `a.b.(c*1000)` and nanopb version `a.b.c.d`
-becomes `a.b.(c*1000+d*10+podspec_revision)`
+| CocoaPods Version | nanopb version |
+| ----------------- | -------------- |
+| 1.0.0             | 0.3.9.5        |
 
-Example: nanopb 0.3.9 => 0.3.9000, and nanopb 0.3.9.1 => 0.3.9010
 
-Converted to four digit version after upgrade from 0.3.9.1 to 0.3.9.4 failed
-and required a rollback.
+#### Version rationale and background
+
+nanopb doesn't follow https://semver.org/. It uses four digit versions and is
+still setting the major version to 0 which in semver implies any update can be
+breaking.
+
+While nanopb tries to follow a variation of semver and updates the fourth digit
+for non-breaking patch updates, the policy is not enforced for binary
+dependencies which can break when source dependencies don't, like the `#define`
+changes in 0.3.9.4.
+
+The versioning scheme is exacerbated by early Firebase releases specifying
```

Comment:
"The versioning scheme" is ambiguous: it's not clear if you're talking about the original versioning scheme or the new one. Also "exacerbated" with the versioning scheme as a subject doesn't quite fit. The original scheme wasn't problematic at all--the only problem was that many Firebase packages misspecified the constraints of their dependencies.